File: front-end/angular/src/value-objects/prediction.ts
import { IOperation } from "./operation";
import { ITotalPopulation } from "./population";
import { IMedianIncome } from "./income";
import { IRadar } from "./radar";

export interface IPrediction {
    readonly operation: IOperation;
    readonly totalPopulation: ITotalPopulation;
    readonly medianIncome: IMedianIncome;
    readonly competitorCount: number;
    readonly aiInsight: string;
    readonly radar: IRadar;
    readonly isSuccess: boolean;
}

export class Prediction implements IPrediction {
    constructor(
        public readonly operation: IOperation,
        public readonly totalPopulation: ITotalPopulation,
        public readonly medianIncome: IMedianIncome,
        public readonly competitorCount: number,
        public readonly aiInsight: string,
        public readonly radar: IRadar,
        public readonly isSuccess: boolean
    ) { }
}

File: back-end/domain/value_objects/prediction.py
from dataclasses import dataclass
from abc import ABC
from domain.value_objects.population import ITotalPopulation
from domain.value_objects.income import IMedianIncome
from domain.value_objects.radar import IRadar
from domain.value_objects.operation import IOperation

@dataclass(frozen=True)
class IPrediction(ABC):
    operation: IOperation
    total_population: ITotalPopulation
    median_income: IMedianIncome
    competitor_count: int
    ai_insight: str
    radar: IRadar
    is_success: bool

@dataclass(frozen=True)
class Prediction(IPrediction):
    operation: IOperation
    total_population: ITotalPopulation
    median_income: IMedianIncome
    competitor_count: int
    ai_insight: str
    radar: IRadar
    is_success: bool